src/main/java/runhistoryplus/patches/ScoreBreakdownRunHistoryPatch.java

```diff
@@ -0,0 +1,176 @@
+            ReflectionHacks.privateMethod(Metrics.class, "addData", Object.class, Object.class)
+                    .invoke(__instance, "score_breakdown", scoreBreakdown);
+        }
+    }
+
+    @SpirePatch(clz = RunHistoryScreen.class, method = SpirePatch.CLASS)
+    public static class ScoreHitboxField {
+        public static final SpireField<Hitbox> scoreHitbox = new SpireField<>(() -> null);
+    }
+
+    @SpirePatch(clz = RunHistoryScreen.class, method = SpirePatch.CONSTRUCTOR)
+    public static class InitializeScoreHitbox {
+        @SpirePostfixPatch
+        public static void initializeScoreHitbox(RunHistoryScreen __instance) {
+            ScoreHitboxField.scoreHitbox.set(__instance, new Hitbox(0, 0));
+        }
+    }
+
+    @SpirePatch(clz = RunHistoryScreen.class, method = "renderRunHistoryScreen")
+    public static class DisplayScoreBreakdown {
+        @SpireInsertPatch(locator = Locator.class, localvars = {"header1x", "yOffset", "scoreLineXOffset", "scoreText"})
+        public static void displayScoreBreakdown(RunHistoryScreen __instance, SpriteBatch sb, float header1x, @ByRef float[] yOffset, float scoreLineXOffset, String scoreText) {
+            scoreHeading = scoreText;
+            Hitbox scoreHitbox = ScoreHitboxField.scoreHitbox.get(__instance);
+            float width = scoreLineXOffset - header1x;
+            scoreHitbox.resize(width, 40.0F);
+            scoreHitbox.move(header1x + width / 2.0F, yOffset[0]);
+            scoreHitbox.render(sb);
+        }
+
+        public static class Locator extends SpireInsertLocator {
+            public int[] Locate(CtBehavior ctMethodToPatch) throws CannotCompileException, PatchingException {
+                Matcher matcher = new Matcher.MethodCallMatcher(ModIcons.class, "hasMods");
+                return LineFinder.findInOrder(ctMethodToPatch, matcher);
+            }
+        }
+    }
+
+    @SpirePatch(clz = RunHistoryScreen.class, method = "update")
+    public static class DisplayScoreBreakdownTooltip {
+        @SpirePostfixPatch
+        public static void displayScoreBreakdownTooltip(RunHistoryScreen __instance) throws NoSuchFieldException, IllegalAccessException {
+            Hitbox hb = ScoreHitboxField.scoreHitbox.get(__instance);
+            hb.update();
+            String text = getTooltipText(__instance);
+            if (hb.hovered && text != null) {
+                CardCrawlGame.cursor.changeType(GameCursor.CursorType.INSPECT);
+                float tipX = hb.x;
+                float tipY = hb.y - 40.0F * Settings.scale;
+                TipHelper.renderGenericTip(tipX, tipY, scoreHeading, text);
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static String getTooltipText(RunHistoryScreen screen) throws NoSuchFieldException, IllegalAccessException {
+        RunData runData = ReflectionHacks.getPrivate(screen, RunHistoryScreen.class, "viewedRun");
```

Comment:
One additional check to add here:
```
if (runData == null) {
	return null;
}
```

This is needed to prevent the game from crashing if the user applies filters that filter out all runs (e.g., if you never played a daily challenge, and filter down to only daily challenges runs).